src/transactions/TransactionEngineTest.java

```diff
@@ -0,0 +1,79 @@
+package transactions;
+import java.io.IOException;
+import java.util.Map;
+import java.io.File;
+
+/**
+ * TransactionEngineTest is used to test both the generateID() method in the Person
+ * class to ensure that a random 24 alpha numeric transaction ID is created for each person
+ * and also the getSchema() and getPersonMap() methods in the DataManager class
+ */
+public class TransactionEngineTest {
+    public static void main(String[] args) {
+        //create a few Person objects to use while testing
+        //this Person object will have every field filled to make sure that a transaction ID is created for a normal Person object
+        Person person1 = new Person("1564afdaave74", "LLBean", "John", "Doe", "555 A Street", "Apartment 1", "Framingham", "Massachusetts", "01701");
+        //these Person objects will have several null fields and I want to make sure a transaction ID is still created
+        Person person2 = new Person("0000000011", "Hogwarts", "Ron", "Weesley", "4 Wizard Way", null, "London", null, null);
+        Person person3 = new Person("afdaea5481dd0", null, "Jane", "Biles", "1 Russell Street", null, null, null, "01721");
+        //this Person object will have several empty strings as fields and I want to make sure a transaction ID is still created
+        Person person4 = new Person("123456451aefa0", "Barnes and Noble", "", "Austen", "444 Pride and Prejudice Way", "", "London", "", "");
+        //this Person object will have several white spaces as fields and I want to make sure a transaction ID is still created
+        Person person5 = new Person("1564affdalvje", "Central Perk", "Rachel", "Green", "   ", "    ", "New York City", "New York", "  ");
+        //this Person object will have a mix of null, empty strings, and white space and I want to make sure a transaction ID is still created
+        Person person6 = new Person("01721aefdeesfe", "   ", null, "Parker", "", null, "Miami", "Florida", "");
+
```

Comment:
I love these

Reply:
Yeah, I got kinda bored so I started making people based off of movies and tv shows